src/neo-vm/Unsafe.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Neo.VM
+{
+    internal static class Unsafe
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static bool Equals(byte[] x, byte[] y)
+        {
+            if (ReferenceEquals(x, y)) return true;
+            if (x is null || y is null) return false;
+            int len = x.Length;
+            if (len != y.Length) return false;
+            fixed (byte* xp = x, yp = y)
+            {
+                long* xlp = (long*)xp, ylp = (long*)yp;
+                for (; len >= 8; len -= 8)
+                {
+                    if (*xlp != *ylp) return false;
+                    xlp++;
+                    ylp++;
+                }
+                byte* xbp = (byte*)xlp, ybp = (byte*)ylp;
+                for (; len > 0; len--)
+                {
+                    if (*xbp != *ybp) return false;
+                    xbp++;
+                    ybp++;
+                }
+            }
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static void MemoryCopy(byte[] src, int srcOffset, byte[] dst, int dstOffset, int count)
+        {
+            fixed (byte* sp = &src[srcOffset], dp = &dst[dstOffset])
+            {
+                Buffer.MemoryCopy(sp, dp, dst.Length - dstOffset, count);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static bool NotZero(byte[] x)
+        {
+            if (x is null)
+                throw new ArgumentNullException(nameof(x));
+            int len = x.Length;
+            if (len == 0) return false;
+            fixed (byte* xp = x)
+            {
+                long* xlp = (long*)xp;
```

Comment:
We need to check if the length is more than 8 bytes, otherwise we are taking random memory

Reply:
https://github.com/neo-project/neo-vm/blob/81c3358a5b8c1611321527cd790465417e4a5cbc/src/neo-vm/Unsafe.cs#L54

This line has checked it.

Reply:
But first is converted to `long`

Reply:
Not convert to `long`, but the `long*`, the pointer.

Reply:
Yes, i tested, and it works as you expected!